src/tests/fake/indexes.ts

```diff
@@ -0,0 +1,78 @@
+import { faker } from "@faker-js/faker";
+import { merge } from "lodash";
+import nock from "nock";
+import { IndexMinimal, IndexNested } from "../../js/indexes/types";
+import { JobMinimal } from "../../js/jobs/types";
+import { ReferenceNested } from "../../js/references/types";
+import { UserNested } from "../../js/users/types";
+import { createFakeJobMinimal } from "./jobs";
+import { createFakeReferenceNested } from "./references";
+import { createFakeUserNested } from "./user";
+import { FakeSearchResults } from "./utils";
+
+type CreateFakeIndexNestedProps = {
+    id?: string;
+    version?: number;
+};
+
+export function createFakeIndexNested(props?: CreateFakeIndexNestedProps): IndexNested {
+    const defaultIndexNested = {
+        id: faker.random.alphaNumeric(8),
+        version: faker.datatype.number(5),
+    };
+
+    return merge(defaultIndexNested, props);
```

Comment:
Yeah it basically does, but if props is undefined it throws an error. This could be easily fixed by assigning an empty object to props if nothing is passed. But this soln reads more clearly to me